### DISCUSSION TITLE
Run `borg check` on large remote repository

### DISCUSSION THREAD

**Discussion by felixschndr** [borgbackup/borg#8301] [Q&A] [answered]
I was just reading a little about `borg check`. I was wondering if it makes sense to run `borg check` on the remote server if the repository is remote and considerably large. I guess borg check (especially when using `--verify-data`) needs to access all the data of the repo. This could be hundreds of GBs which would take a long time to transfer over the Internet. Would it be sensible to run `borg check` on the remote side to not have to transfer all that data? And what amounts of data am I looking at when **not** using `--verify-data`?

**Comment by ThomasWaldmann** [comment]
Do you run borg as client/server (== do you use a ssh://... repo URL)?

**Reply by felixschndr** [reply]
Yes, I do (should have specified that in the original post, my mistake)

**Reply by ThomasWaldmann** [reply, marked as answer]
OK, for client/server operation of `borg check`, it is:
- running the repository part of the check on the server (crc32 checking of segments), thus usually not much network I/O.
- running the archives part of the check on the client (needs some network I/O to check the archive metadata). same for `--verify-data` (checks file content data chunks, needs potentially lots of additional network I/O [guess as big as the repo] and time for that).

Starting `borg check` on the server:
- does not change much for the repo part, as that runs on the server anyway, even if you invoke the command on the client.
- for the archives part of borg check, it needs to be able to decrypt data. would require the borg key on the server and you entering its passphrase on the server. depending on whether the server can be trusted that can be an option or not.
